Tests/ReactiveSwiftTests/SchedulerSpec.swift

```diff
@@ -218,6 +218,34 @@ class SchedulerSpec: QuickSpec {
 					scheduler.queue.resume()
 					expect{count}.toEventually(equal(timesToRun))
 				}
+				
+				it("should repeatedly run actions after a given date when the disposable is not retained") {
+					let disposable = SerialDisposable()
+					
+					var count = 0
+					let timesToIncrement = 3
+					
+					// Start two repeating timers, dispose the first, and ensure only the second runs.
+					
+					disposable.inner = scheduler.schedule(after: Date(), interval: .milliseconds(10), leeway: .seconds(0)) {
+						expect(false).to(equal(true), description: "timer not cancelled on disposal")
+					}
+					
+					scheduler.schedule(after: Date(), interval: .milliseconds(10), leeway: .seconds(0)) {
```

Comment:
You might scope this in a function, so that the disposable is guaranteed to be deinitialised.